### ISSUE TITLE
Why we don't need to use _offset_image function

### ISSUE THREAD

**Issue by JunbinWang** [jonrei/tf-AdaIN#8]
Hi,

In AdaIN.py file, line 93 and line 99, you comment out the subtract mean value code. Can you explain why you don't need to subtract the mean value for the input image ?

**Comment by jonrei** [comment]
Hi,
If I remember correctly, my thoughts were that there was a bug in the original Torch implementation. This means that in the original implementation, they have trained the model without subtracting the means, which in turns makes the output from using mean-adjusted images worse.
Maybe better results can be yielded by retraining the model with adjusted means, but I never got around to try it.

**Comment by JunbinWang** [comment]
COOL. Thank you very much for your reply!
I am trying to re-implement it recently. 
I was thinking if it is because the Vgg-19 model we use is normalized, which means the input value of the image is also normalised to the range between 0 and 1, so we don't need to substract the original mean values ....

**Comment by eridgd** [comment]
Hey @JunbinWang @jonrei, sorry to interject, but I actually have an answer for this! 

I implemented [my own version](https://github.com/eridgd/AdaIN-TF) of AdaIN in TF with the `graph_from_t7` function borrowed from this repo and I ran into the same weird question of why it wasn't subtracting the means. That was answered by printing the layers loaded from `vgg_normalised.t7` and seeing that the first layer is a 1x1 SpatialConvolution with kernel weights:

```
'weight': array([[[[  0.]],

        [[  0.]],

        [[255.]]],


       [[[  0.]],

        [[255.]],

        [[  0.]]],


       [[[255.]],

        [[  0.]],

        [[  0.]]]])
```

and bias:

```'bias': array([-103.939, -116.779, -123.68 ])```


So this layer is handling the preprocessing in-network by multiplying the channels in [0,1] by 255 then adding the negative BGR means as bias. The rest of the network is regular VGG19 with reflection padding instead of zeros (see [here](https://github.com/xunhuang1995/AdaIN-style/blob/master/models/download_models.sh)). Pretty clever, no?

Also, the original caffe model is taken from [here](http://bethgelab.org/deeptextures/) and what they mean by normalization is:

> The weights in the normalised network are scaled such that the mean activation of each filter over images and positions is equal to one. Such re-scaling can always be done without changing the output of a neural network as long as the non-linearities in the network is are rectifying linear.
